llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -39,6 +47,30 @@ def from_str(
             role = MessageRole(role)
         return cls(role=role, content=content, **kwargs)
 
+    def _recusrive_serialization(self, value: Any) -> Any:
+        if isinstance(value, (V1BaseModel, V2BaseModel)):
+            return value.dict()
+        if isinstance(value, dict):
+            return {
+                key: self._recusrive_serialization(value)
+                for key, value in value.items()
+            }
+        if isinstance(value, list):
+            return [self._recusrive_serialization(item) for item in value]
+        return value
+
+    def dict(self, **kwargs: Any) -> dict:
+        # ensure all additional_kwargs are serializable
+        msg = super().dict(**kwargs)
+
+        for key, value in msg["additional_kwargs"].items():
+            value = self._recusrive_serialization(value)
+            if not isinstance(value, (str, int, float, bool, dict, list, type(None))):
+                value = str(value)
```

Comment:
should we still throw an error here? a bit worried it'll lead to unexpected behavior when deserializing

Reply:
yea thats fair